build/commands/lib/util.js

```diff
@@ -65,6 +66,8 @@ async function applyPatches() {
     process.exit(1)
   }
 
+  await updateUnsafeBuffersPaths()
```

Comment:
can we write this somewhere else, call it in run hooks and override `clang_unsafe_buffers_paths`?

Reply:
cc: @cdesouza-chromium

Reply:
After discussions with @bridiver I'm considering having this running as a hook. @goodov Any thoughts?

Reply:
> override `clang_unsafe_buffers_paths`

That's fine, but not sure if this should be part of hooks. I'd rather keep this code in build scripts so the path to a generated file is shared between the code that generates it and the arg we pass to GN.

Reply:
The value can be shared through gclient_args.gni. We used to do the same thing with the rust version in our old rust build process

Reply:
> The value can be shared through gclient_args.gni. We used to do the same thing with the rust version in our old rust build process

`gclient_args.gni` (or `brave_gclient_args.gni`) should be imported at the right place to override `clang_unsafe_buffers_paths`. We either need to patch `BUILDCONFIG.gn` or import the generated file manually via `args_generated.gni`.

I agree we should make use of `clang_unsafe_buffers_paths` in this PR, but a possible move to hooks looks like a follow up to me.

Reply:
Fixed in 72935603d2077749971b9da2a8efa8bdb274c064. This change makes use of the gn arg `clang_unsafe_buffers_paths`. An issue has been created as a follow up to move this code to `runhooks` https://github.com/brave/brave-browser/issues/38512